lib/OpenQA/CLI/schedule.pm

```diff
@@ -0,0 +1,97 @@
+sub _monitor_jobs ($self, $client, $poll_interval, $job_ids, $job_results) {
+    while (@$job_results < @$job_ids) {
+        my $job_id = $job_ids->[@$job_results];
+        my $tx = $client->start($client->build_tx(GET => $self->url_for("experimental/jobs/$job_id/status"), {}));
+        my $res = $self->handle_result($tx, {pretty => 0, quiet => 1, links => 0, verbose => 0});
+        return $res if $res != 0;
+        my $job = $tx->res->json;
+        my $job_state = $job->{state} // NONE;
+        if (OpenQA::Jobs::Constants::meta_state($job_state) eq OpenQA::Jobs::Constants::FINAL) {
+            push @$job_results, $job->{result} // NONE;
+            next;
+        }
+        print "Job state of job ID $job_id: $job_state, waiting…\n";
+        sleep $poll_interval;
+    }
+}
+
+sub _compute_return_code ($self, $job_results) {
+    for my $job_result (@$job_results) {
+        return 2 unless OpenQA::Jobs::Constants::is_ok_result($job_result);
+    }
+    return 0;
+}
+
+sub command ($self, @args) {
+    die $self->usage
+      unless getopt \@args,
+      'param-file=s' => \my @param_file,
+      'm|monitor' => \my $monitor,
+      'i|poll-interval=i' => \my $poll_interval,
+      'r|set-return-code' => \my $set_return_code;
+    @args = $self->decode_args(@args);
+    my $client = $self->client($self->post_url);
+
+    my @job_ids;
+    my $create_res = $self->_create_jobs($client, \@args, \@param_file, \@job_ids);
+    return $create_res if $create_res != 0 || !($monitor || $set_return_code);
+
+    my @job_results;
+    my $monitor_res = $self->_monitor_jobs($client, $poll_interval // 10, \@job_ids, \@job_results);
+    return $monitor_res if $monitor_res != 0 || !$set_return_code;
+    return $self->_compute_return_code(\@job_results);
+}
+
+1;
+
+=encoding utf8
+
+=head1 SYNOPSIS
+
+  Usage: openqa-cli schedule [OPTIONS] DISTRI=… VERSION=… FLAVOR=… ARCH=… [ISO=… …]
+
+  Options:
+        --apibase <path>           API base, defaults to /api/v1
+        --apikey <key>             API key
+        --apisecret <secret>       API secret
+        --host <host>              Target host, defaults to http://localhost
+    -h, --help                     Show this summary of available options
+        --osd                      Set target host to http://openqa.suse.de
+        --o3                       Set target host to https://openqa.opensuse.org
+        --param-file <param=file>  Load content of params from files instead of
+                                   from command line arguments. Multiple params
+                                   may be specified by adding the option
+                                   multiple times
+    -m, --monitor                  Wait until all jobs are done/cancelled
+    -i, --poll-interval            Specifies the poll interval used with --monitor
+    -r, --set-return-code          Return non-zero exit code if at least one job
+                                   is not passed/softfailed (implies -m)
```

Comment:
Why not make that the default for `-m` anyway?

Reply:
I don't like scripts that do too much at the same time, at least not by default.

It should also be optional so this sub command is usable in more general way. However, I prefer positive arguments over negative ones which speaks also for not making it a default.

It is also an unintuitive default for a client to return a non-zero code if the request generally worked. (If I request a page with the contents "failure" I don't expect curl to return a non-zero code for this.)

Reply:
yeah but for `monitor` this is something that could be expected and matches our use case if the exit code shows the result of monitored jobs. If you want to avoid negations then maybe cli should return 2 on request errors, not 1 :)

Reply:
Can you clarify the use cases of "-r" versus "-rm"? I feel like I would always want the return code. Not sure when it's useful not to have 🤔